includes/class-main.php

```
@@ -209,6 +212,65 @@ public function __construct() {
+					$user = wp_get_current_user();
+					if ( ! empty( $user->roles ) ) {
+						$roles = $user->roles;
+						sort( $roles );
+						$hash = substr( md5( implode( ',', $roles ) ), 0, 12 );
+						if ( ! isset( $_COOKIE['wppo_role_hash'] ) || $_COOKIE['wppo_role_hash'] !== $hash ) {
+							setcookie( 'wppo_role_hash', $hash, time() + DAY_IN_SECONDS, COOKIEPATH, COOKIE_DOMAIN, is_ssl(), true );
+						}
+					}
+				}
+			}
+		}
```

PHPMD (2.15.0)

[error] set_role_hash_cookie accesses the super-global variable $_COOKIE. (undefined)

(Superglobals)